plutus-tx-plugin/src/PlutusTx/Plugin.hs

```diff
@@ -119,14 +122,18 @@ plugin = GHC.defaultPlugin { GHC.pluginRecompile = GHC.flagRecompile
              : rest
 
 -- | A simplifier pass, implemented by GHC
-mkSimplPass :: GHC.DynFlags -> GHC.CoreToDo
-mkSimplPass flags =
+mkSimplPass :: GHC.DynFlags -> GHC.Logger -> GHC.CoreToDo
+mkSimplPass flags logger =
   -- See Note [Making sure unfoldings are present]
   GHC.CoreDoSimplify 1 $ GHC.SimplMode {
               GHC.sm_names = ["Ensure unfoldings are present"]
             , GHC.sm_phase = GHC.InitialPhase
+            , GHC.sm_uf_opts = GHC.defaultUnfoldingOpts
             , GHC.sm_dflags = flags
             , GHC.sm_rules = False
+            , GHC.sm_cast_swizzle = True
+            , GHC.sm_pre_inline = True
```

Comment:
do we definitely want these? our default is "don't do things", I think

Reply:
I think these are both enabled by default in GHC 8. If I turn off `sm_pre_inline`, I got `E042:Error: Unsupported feature: Type constructor: GHC.Prim.Char#` in one of the test cases. I'll create a separate ticket for investigating in this.

Reply:
Okay, also add a comment in the code pointing to the issue please.